addons/project/models/project_task_recurrence.py

```diff
@@ -58,286 +21,94 @@ class ProjectTaskRecurrence(models.Model):
-        if self.filtered(lambda r: r.repeat_type == 'until' and r.repeat_unit == 'month' and r.repeat_until and r.repeat_on_month == 'date'
-           and int(r.repeat_day) > r.repeat_until.day and monthrange(r.repeat_until.year, r.repeat_until.month)[1] != r.repeat_until.day):
-            raise ValidationError(_('The end date should be after the day of the month or the last day of the month'))
-
     @api.model
-    def _get_recurring_fields(self):
-        return ['message_partner_ids', 'company_id', 'description', 'displayed_image_id', 'email_cc',
-                'parent_id', 'partner_id', 'partner_phone', 'planned_hours',
-                'project_id', 'display_project_id', 'project_privacy_visibility', 'sequence', 'tag_ids', 'recurrence_id',
-                'name', 'recurring_task', 'analytic_account_id', 'user_ids']
-
-    def _get_weekdays(self, n=1):
-        self.ensure_one()
-        if self.repeat_unit == 'week':
-            return [fn(n) for day, fn in DAYS.items() if self[day]]
-        return [DAYS.get(self.repeat_weekday)(n)]
+    def _get_recurring_fields_to_copy(self):
+        return [
+            'analytic_account_id',
+            'company_id',
+            'description',
+            'displayed_image_id',
+            'email_cc',
+            'message_partner_ids',
+            'name',
+            'parent_id',
+            'partner_id',
+            'partner_phone',
+            'planned_hours',
+            'project_id',
+            'display_project_id',
+            'project_privacy_visibility',
+            'recurrence_id',
+            'recurring_task',
+            'sequence',
+            'tag_ids',
+            'user_ids',
+        ]
 
     @api.model
-    def _get_next_recurring_dates(self, date_start, repeat_interval, repeat_unit, repeat_type, repeat_until, repeat_on_month, repeat_on_year, weekdays, repeat_day, repeat_week, repeat_month, **kwargs):
-        count = kwargs.get('count', 1)
-        rrule_kwargs = {'interval': repeat_interval or 1, 'dtstart': date_start}
-        repeat_day = int(repeat_day)
-        start = False
-        dates = []
-        if repeat_type == 'until':
-            rrule_kwargs['until'] = repeat_until if repeat_until else fields.Date.today()
-        else:
-            rrule_kwargs['count'] = count
-
-        if repeat_unit == 'week'\
-            or (repeat_unit == 'month' and repeat_on_month == 'day')\
-            or (repeat_unit == 'year' and repeat_on_year == 'day'):
-            rrule_kwargs['byweekday'] = weekdays
+    def _get_recurring_fields_to_postpone(self):
+        return [
+            'date_deadline',
+        ]
+
+    def _get_last_task_id_per_recurrence_id(self):
+        return {} if not self else {
+            record['recurrence_id'][0]: record['max_id']
+            for record in self.env['project.task']._read_group(
+                domain=[
+                    ('recurrence_id', '!=', False),
+                    ('recurrence_id', 'in', self.ids),
```

Comment:
```suggestion
                    ('recurrence_id', 'in', self.ids),
```
Unless I'm missing something, I don't think the first condition is useful?

Reply:
true, that's why I added the first line in the method.